src/java.sql/share/classes/java/sql/DriverManager.java

```diff
@@ -654,7 +654,7 @@ private static Connection getConnection(
         /*
          * When callerCl is null, we should check the application's
          * (which is invoking this class indirectly)
-         * classloader, so that the JDBC driver class outside rt.jar
+         * classloader, so that the JDBC driver class outside the image
          * can be loaded from here.
```

Comment:
This code should probably be changed to use VM.isSystemDomainLoader(callerCL).

I think the comment should be replaced because it doesn't match what it actually does and it's nothing to do with the whether the JDBC driver is in the run-time image or not. How about:

"If the caller is defined to the bootstrap or platform class loader then use the Thread CCL as the initiating class loader so that a JDBC on the class path, or bundled with an application, is found."

Reply:
Hi Alan, regarding usage of class VM I get 
'package jdk.internal.misc is declared in module java.base, which does not export it to module java.sql'
Is there any concern to export it as well to module java.sql  ?
And btw did you mean to use it like this, in the if  ?

`        if (callerCL == null || VM.isSystemDomainLoader(callerCL)) {
            callerCL = Thread.currentThread().getContextClassLoader();
        }`

Reply:
> Hi Alan, regarding usage of class VM I get 'package jdk.internal.misc is declared in module java.base, which does not export it to module java.sql' Is there any concern to export it as well to module java.sql ? And btw did you mean to use it like this, in the if ?
> 
> `if (callerCL == null || VM.isSystemDomainLoader(callerCL)) { callerCL = Thread.currentThread().getContextClassLoader(); }`

It was just a passing comment, I didn't meant to suggest changing it as part of this PR. We have always think twice before adding qualified exports from java.base and this is case where java.sql is very "non-core", we don't want to give it any access to java.base internals.

Reply:
Hi Alan, thanks for clarifying.  So I should only adjust the comment, correct ?

Reply:
> Hi Alan, I adjusted the comment in DriverManager.java .

Thanks, the update looks okay.